### PR TITLE
feat: show URL as link when unable to fetch original post

### LINKED ISSUE #621: Improve the behavior when fetching the original author/article failed
Currently, the original author and the link to the original article will be missing when it fails to fetch the original post.
スクリーンショット 2023-04-19 100837

How about at least showing the URL? It's not the best appearance but should be better than nothing.
Like this:
スクリーンショット 2023-04-20 001533

**To Reproduce**
This issue happens when fetching the original post failed during the build.

**Expected behavior**
Show the URL of the original article as a link.

### PR DESCRIPTION
**Pull request by sidemt**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #621

When failed to fetch the original post (show URL as a link):
スクリーンショット 2023-04-28 220314

When successfully fetched the original post (show article title as a link):
スクリーンショット 2023-04-28 220345

### CODE REVIEW

File: utils/ghost/original-post-handler.js [hunk 1]
```diff
@@ -12,6 +12,9 @@ const originalPostHandler = async post => {
   const match = originalPostRegex.exec(headAndFootCode);
 
   if (match) {
+    let href_value = '';
+    let link_text = '';
```
Comment (scissorsneedfoodtoo) [inline utils/ghost/original-post-handler.js]:
```suggestion
    let hrefValue = '';
    let linkText = '';
```

File: utils/ghost/original-post-handler.js [hunk 2]
```diff
@@ -51,26 +54,9 @@ const originalPostHandler = async post => {
...
-        <p data-test-label="translation-intro">
-          ${originalArticleHTML}
-        </p>`;
-
-      // Append details about the original article
-      // to the beginning of the translated article
-      post.html = introHTML + post.html;
+      // Use the title of the original post as link text
+      href_value = originalPost.url;
+      link_text = originalPost.title;
```
Comment (scissorsneedfoodtoo) [inline utils/ghost/original-post-handler.js]:
```suggestion
      hrefValue = originalPost.url;
      linkText = originalPost.title;
```

File: utils/ghost/original-post-handler.js [hunk 3]
```diff
@@ -81,7 +67,32 @@ const originalPostHandler = async post => {
       post has not been deleted.
       ---------------------------------------------------------------
       `);
+
+      // Use URL in the script tag as link text
+      href_value = match.groups.url;
+      link_text = match.groups.url;
```
Comment (scissorsneedfoodtoo) [inline utils/ghost/original-post-handler.js]:
```suggestion
      hrefValue = match.groups.url;
      linkText = match.groups.url;
```

File: utils/ghost/original-post-handler.js [hunk 4]
```diff
@@ -81,7 +67,32 @@ const originalPostHandler = async post => {
...
     }
+
+    const originalArticleHTML = translate(
+      'original-author-translator.details.original-article',
+      {
+        '<0>': '<strong>',
+        '</0>': '</strong>',
+        title: `<a href="${href_value}" target="_blank" rel="noopener noreferrer" data-test-label="original-article-link">${link_text}</a>`,
```
Comment (scissorsneedfoodtoo) [inline utils/ghost/original-post-handler.js]:
```suggestion
        title: `<a href="${hrefValue}" target="_blank" rel="noopener noreferrer" data-test-label="original-article-link">${linkText}</a>`,
```

**Review by scissorsneedfoodtoo — COMMENTED** [review]
Thanks again for all of your help with this @sidemt! Tested locally and everything LGTM.

My only suggestions are very minor ones about a couple of variable names.

I don't know if it's 100% consistent throughout the code, but I tried to only use snake case for variable and property names when working with the Ghost post / page objects directly (reading, appending data, and so on). I've found that it can be helpful to track which data is coming directly from Ghost, or should look like it's coming from Ghost, for use in the templates.

Either way, I think this is a huge improvement.

**Comment by sidemt** [comment]
@scissorsneedfoodtoo Thank you for catching that! I applied your suggestions.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Tested the recent changes locally and everything LGTM 👍 👍

**Pull request merged**